components/style/servo/url.rs

```diff
@@ -126,3 +127,25 @@ impl ToCss for SpecifiedUrl {
         dest.write_str(")")
     }
 }
+
+impl ToComputedValue for SpecifiedUrl {
+    type ComputedValue = ComputedUrl;
+
+    fn to_computed_value(&self, _: &Context) -> Self::ComputedValue {
+        match self.resolved {
+            Some(ref url) => ComputedUrl::Valid(url.clone()),
+            None => ComputedUrl::Invalid(Arc::new("about:invalid".to_string())),
+        }
+    }
+
+    fn from_computed_value(computed: &ComputedUrl) -> Self {
```

Comment:
Also, if you did what I suggest above, you could use `original: None, resolved: Some()` for valid urls, and `original: Some(), resolved: None` for invalid ones, which would match what we do when computing them.